### PR TITLE
feat(product): add canonical URL to product

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the new behavior?
Adds `canonicalUrl` to `DaffProduct`

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
Note that the Magento driver will not output the correct value until [`magento2-daffodil` is updated](https://github.com/graycoreio/magento2-daffodil/issues/3).

### CODE REVIEW

File: libs/product/src/models/product.ts [hunk 1]
```diff
@@ -17,7 +18,7 @@ export enum DaffProductTypeEnum {
 /**
  * An interface for a product usable by the @daffodil/product library.
  */
-export interface DaffProduct extends DaffLocatable, DaffIdentifiable {
+export interface DaffProduct extends DaffLocatable, DaffIdentifiable, Partial<DaffCanonicallyLocatable> {
```
Comment (damienwebdev) [inline libs/product/src/models/product.ts:21]: Why is this a partial?

File: libs/product/driver/magento/src/transforms/spec-data/daff-composite-product.json [hunk 2]
```diff
@@ -2,6 +2,7 @@
 	"type": "composite",
 	"id": "22",
 	"url": "/url.html",
+  "canonicalUrl": "/canonical-url.html",
```
Comment (damienwebdev) [inline libs/product/driver/magento/src/transforms/spec-data/daff-composite-product.json:5]: For correctness, give this a full URL with protocol and domain.

File: libs/product/testing/src/factories/product.factory.ts [hunk 3]
```diff
@@ -17,6 +17,7 @@ export class MockProduct implements DaffProduct {
 	type = DaffProductTypeEnum.Simple;
 	id = faker.datatype.uuid();
 	url = `/${faker.random.word()}.html`;
+	canonicalUrl = `/${faker.random.word()}.html`;
```
Comment (damienwebdev) [inline libs/product/testing/src/factories/product.factory.ts:20]: For correctness, give this a full URL with protocol and domain.

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: libs/product/src/models/product.ts (hunk 1 above)
Comment (griest024) [reply libs/product/src/models/product.ts:21]: Canonical URLs aren't needed (I assume) for product previews so it should be optional.

File: libs/product/driver/magento/src/transforms/spec-data/daff-composite-product.json (hunk 2 above)
Comment (griest024) [reply libs/product/driver/magento/src/transforms/spec-data/daff-composite-product.json]: changed

File: libs/product/testing/src/factories/product.factory.ts (hunk 3 above)
Comment (griest024) [reply libs/product/testing/src/factories/product.factory.ts]: changed

**Review by damienwebdev — APPROVED** [review]

**Pull request merged**